File: bzk/bzkflow/src/main/java/net/bzk/flow/api/dto/RegisteredFlow.java
package net.bzk.flow.api.dto;

import java.util.List;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import net.bzk.flow.model.Flow;
import net.bzk.flow.run.flow.FlowRuner.State;

@Data
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public class RegisteredFlow {

	private Flow flow;
	private List<RunInfo> runInfos;

	@Data
	@AllArgsConstructor(access = AccessLevel.PUBLIC)
	public static class RunInfo {
		private String runUid;
		private State state;
		private String endTag;
		private String endLinkUid;
	}

}

File: flow-editor/src/app/dto/registered-flow.ts
import { Flow } from '../model/flow';

export class RegisteredFlow {

  public flow: Flow;
  public runInfos: Array<RunInfo>;


}

export class RunInfo {

  public runUid: string;
  public state = FlowState.Pedding;
  public endTag: string;
  public endLinkUid: string;

}

export enum FlowState {
  Pedding = 'Pedding', Running = 'Running', Done = 'Done', Fail = 'Fail'
}
